### DISCUSSION TITLE
thread/wangzhe-thread

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#96] [Announcements]
# thread/wangzhe-thread

室友打了一把王者就学会了创建Java线程的3种方式

https://tobebetterjavaer.com/thread/wangzhe-thread.html

**Comment by lindas-cat** [comment]
太笨了我

**Comment by iancxm** [comment]
博主好，你的这个进阶之路给了我很大的帮助。这里有个内容优化建议，可以添加下在继承与实现接口的方式中，为什么接口的实现方式能共享变量，而继承的不能共享变量的知识点。这个问题我多花费了差不多两个小时。

**Reply by sunnfun-0** [reply]
继承的应该也可以共享变量吧

**Reply by DEANOFORPHAN** [reply]
同问

**Reply by lin667z** [reply]
继承中可以有私有变量，会导致不能共享。而接口只能定义常量，所以接口的所有变量都可以与其实现类共享。

**Comment by Ricehahaa** [comment]
🐮
